pkg/api/apiloader.go

```diff
@@ -200,6 +200,10 @@ func (a *Apiloader) LoadContainerServiceForAgentPoolOnlyCluster(
 		if e := json.Unmarshal(contents, &managedCluster); e != nil {
 			return nil, IsSSHAutoGenerated, e
 		}
+		// verify managedCluster
+		if managedCluster == nil || managedCluster.Properties == nil {
+			return nil, false, a.Translator.Errorf("properties object in managed cluster should not be nil")
```

Comment:
@shrutir25 actually the properties can be nil and is a supported ARM scenario. RPs need to be able to handle this.

Reply:
Here is the documentation: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#put-resource -


```
properties | Optional. Format not defined by ARM. Settings used to provision or configure the resource. All Resource Provider specific properties must be placed within this. The order of parameters in the request is unspecified. RPs should not rely on any particular ordering.
-- | --
```

Reply:
Here is the documentation: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#put-resource -

```
properties | Optional. Format not defined by ARM. Settings used to provision or configure the resource. All Resource Provider specific properties must be placed within this. The order of parameters in the request is unspecified. RPs should not rely on any particular ordering.
-- | -
```

Reply:
@amanohar - so then you are suggesting that we make this check in RP before calling this function in acs-engine right ?

Reply:
@shrutir25 I meant RP has to support a payload with nil `properties` and PUT should not fail.

Reply:
I talked to Gaurav to get clarification. So long as the fields inside `Properties` are required, we can return 400 if it's not supplied.

Reply:
In case of update correct behavior is to accept `properties`=nil as user is not required to supply `required` properties during update.